### PR TITLE
[🚀] NT-586 Clicking Editorial in disco opens Editorial projects list

### PR DESCRIPTION
**Pull request by eoji**
# 📲 What
Clicking an `Editorial` tout in Discovery now opens the Editorial projects list (`EditorialActivity`)

# 🤔 Why
So we can get some of that sweet editorial content into the app.

# 🛠 How
- Updated strings and config.
- Passing in an `Editorial` to the `EditorialActivity` loads a `DiscoveryFragment` with the `tag_id` of the `Editorial`, sorted by `home` (magic).
- Added `tagId` to `DiscoveryParams`.
- Deleted `go_rewardless_tout` since now `go_rewardless_header` is used in both the tout and the header.
- Tests

## UI
- Synced with Colleen on the spacing of the `Activity` touts. 
- Added `trust_700`. We're slowly transitioning to the KDS palette.

## a11y
- The entire `Editorial` card is `focusable` so both the title and description are read as one item. And since it's `clickable`, TalkBack™ will say "Double tap to activate".

# 👀 See
| After 🦋 | Before 🐛 |
| --- | --- |
| screenshot-2019-11-21_164634 | screenshot-2019-11-21_164625 |
| Editorial with transition | Collapsible toolbar |
| device-2019-11-21-204250 2019-11-21 20_43_52 | device-2019-11-21-204341 2019-11-21 20_44_32 |

# 📋 QA
Smash that editorial (if your feature flag is enabled or you've toggled the switch on Internal builds)

# Story 📖
[NT-586]

[NT-586]: https://dripsprint.atlassian.net/browse/NT-586

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/data/Editorial.kt [hunk 1]
```diff
@@ -4,22 +4,23 @@ import androidx.annotation.ColorRes
...
     companion object {
         val GO_REWARDLESS = Editorial(R.string.Back_it_because_you_believe_in_it,
                 R.string.Find_projects_that_speak_to_you,
                 R.string.Join_us_in_supporting_creative_work_for_its_own_sake_this_holiday_season,
-                R.string.Use_the_hashtag_hashtag_backeditbecause_to_share_what_projects_youre_supporting_and_why_theyre_important_to_you,
+                R.string.These_projects_could_use_your_support,
                 R.color.trust_700,
-                R.drawable.go_rewardless_tout,
+                R.drawable.go_rewardless_header,
                 518)
```
[the remark this answers is not in the document]
Comment (eoji) [reply app/src/main/java/com/kickstarter/ui/data/Editorial.kt]: Yeah `Editorial` is just a data class, so I use `editorial.tag_id`. I can make the `Editorial` class an enum!`518` will still be hardcoded though. Does that make sense?

File: app/src/main/java/com/kickstarter/viewmodels/EditorialViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        private val title: BehaviorSubject<Int> = BehaviorSubject.create()
+
+        private val apiClient: ApiClientType = environment.apiClient()
+
+        val inputs: Inputs = this
+        val outputs: Outputs = this
+
+        init {
+            val editorial = intent()
+                    .map { it.getSerializableExtra(IntentKey.EDITORIAL) }
```
[the remark this answers is not in the document]
Comment (eoji) [reply app/src/main/java/com/kickstarter/viewmodels/EditorialViewModel.kt]: So an [`Intent`](https://developer.android.com/reference/android/content/Intent) is a way that Android components pass data to each other. It contains a [`Bundle`](https://developer.android.com/reference/android/os/Bundle.html) (fancy hashmap) of "extras". So I'm pulling the `Editorial` object out of the extras with the key `IntentKey.EDITORIAL`.

File: app/src/main/java/com/kickstarter/ui/data/Editorial.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinswart) [reply app/src/main/java/com/kickstarter/ui/data/Editorial.kt]: Sure, I guess in our case we refer to it in several places so the enum just made it easier. Whatever works for you!

File: app/src/test/java/com/kickstarter/viewmodels/EditorialViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        this.rootCategories.assertValue(listOf(CategoryFactory.artCategory(),
+                CategoryFactory.gamesCategory()))
+    }
+
+    @Test
+    fun testTitle() {
+        setUpEnvironment(environment(), Editorial.GO_REWARDLESS)
+
+        this.title.assertValue(R.string.Join_us_in_supporting_creative_work_for_its_own_sake_this_holiday_season)
```
Comment (justinswart) [inline app/src/test/java/com/kickstarter/viewmodels/EditorialViewModelTest.kt]: Oh lol, this is the string now? The figma designs still have the other one 🤔

Comment (eoji) [reply app/src/test/java/com/kickstarter/viewmodels/EditorialViewModelTest.kt]: At one point in time it was 😭 I'll change it

**Pull request merged**